New Guides section, Serverless as an evaluation and prototyping platform (#5118)
influxdata/docs-v2: content/influxdb/cloud-serverless/guides/prototype-evalutation.md

  - [Query Language Differences](#query-language-differences)
  - [API and client library differences](#api-and-client-library-differences)
  - [Tasks and alerts differences](#tasks-and-alerts-differences)
  - [Token management and authorization differences](#token-management-and-authorization-differences)
  - [Performance differences](#performance-differences)
  - [Schema differences](#schema-differences)
  - [Organization management differences](#organization-management-differences)
- [Best practices](#best-practices)
  - [Use the v3 lightweight client libraries](#use-the-v3-lightweight-client-libraries)
  - [Avoid features that are not included with InfluxDB Cloud Dedicated](#avoid-features-that-are-not-included-with-influxdb-cloud-dedicated)
  - [Use SQL or InfluxQL as your Query Language](#use-sql-or-influxql-as-your-query-language)
  - [Stay within the schema limits of InfluxDB Cloud Serverless](#stay-within-the-schema-limits-of-influxdb-cloud-serverless)
  - [Keep test and production data separate](#keep-test-and-production-data-separate)
  
<!-- END TOC -->

## Key differences between InfluxDB Cloud Serverless and Cloud Dedicated

### User interface differences

> The user interfaces between InfluxDB Cloud Serverless and InfluxDB Cloud
> Dedicated are completely different.

Reviewer: ```suggestion
Administrative tools and user interfaces differ between InfluxDB Cloud Serverless and InfluxDB Cloud
Dedicated.
```

- [Send alerts using data in InfluxDB Cloud Serverless](/influxdb/cloud-serverless/process-data/send-alerts/)
- [Downsample data](/influxdb/cloud-serverless/process-data/downsample/)
- [Summarize data](/influxdb/cloud-serverless/process-data/summarize/)
- [Use data analysis tools](/influxdb/cloud-serverless/process-data/tools/)

### Token management and authorization differences

In addition to the token management UI differences mentioned previously
(there is a UI and API for this with Cloud Serverless, with InfluxDB Cloud
Dedicated you use `influxctl`), there are also some differences in the granularity
of token permissions---InfluxDB Cloud Dedicated has a few more permission options. 

| Function             | InfluxDB Cloud Serverless | InfluxDB Cloud Dedicated |
| :------------------- | :------------------------ | :----------------------- |
| API token generation | Built-in UI or v2 API     | `influxctl` CLI          |

### Performance differences

InfluxDB Cloud Serverless is a multi-tenant solution and has a number of factors
that could affect database performance, including:

- **Rate limiting**: limits on reads and writes in InfluxDB Cloud Serverless

> - **Noisy neighbors**: unexpected high usage from other customers could cause
>   slight performance differences in your own experience.
> - **Tag-specific queries**: query performance is optimized generically with

Author: ```suggestion
- **Noisy neighbors**: Unexpected high usage from other customers could cause
  slight performance differences in your own experience.
- **Tag-specific queries**: Query performance is optimized generically with
```

  of tag values may be noticeably slower compared to previous versions.
- **Generic optimization**: Cloud Serverless is generically optimized to fit the
  most common workloads. InfluxDB Cloud Dedicated and can be tuned to your
  specific data sets and workload with options such as custom partitioning and others.

If you are using InfluxDB Cloud Serverless for evaluation purposes, consider
investing in an actual proof-of-concept (PoC) of InfluxDB Cloud Dedicated to get
a more accurate picture of your expected performance.

### Schema differences

Schema support in InfluxDB Cloud Serverless and InfluxDB Cloud Dedicated is basically identical.
However, InfluxDB Cloud Dedicated provides more flexibility by letting you raise
or exceed the normal limits for the number of tables (measurements) and columns
(time, fields, and tags) in a database.
_See [Stay within the schema limits of InfluxDB Cloud Serverless](#stay-within-the-schema-limits-of-influxdb-cloud-serverless)_.

### Organization management differences

InfluxDB Cloud Serverless has a feature that lets you create sub-organizations
within your account. InfluxDB Cloud Dedicated does not have this feature.

> Additionally, due to being multi-tenant, Cloud Serverless requires the use of an
> organization name or organization ID in many interactions where this is not
> required in Cloud Dedicated.

Reviewer: ```suggestion
Additionally, due to being multi-tenant, Cloud Serverless requires specifying
organization name or ID in some interactions. This isn't required with Cloud Dedicated.
```

## Best practices

> The following are recommended best practices for using InfluxDB Cloud Serverless
> as an evaluation or prototyping platform for InfluxDB Cloud Dedicated.

Reviewer: ```suggestion
Follow these recommended best practices when using InfluxDB Cloud Serverless
as an evaluation or prototyping platform for InfluxDB Cloud Dedicated.
```

### Use the v3 lightweight client libraries

> Use the InfluxDB [v3 lightweight client libraries](/influxdb/cloud-serverless/reference/client-libraries/v3/)
> to ensure your application code that writes to and queries InfluxDB will be
> equally compatible with InfluxDB Cloud Serverless and Dedicated.
> The only change you’ll need to make is to your InfluxDB connection credentials
> (host, database name, and token).

Reviewer: ```suggestion
Use the InfluxDB [v3 lightweight client libraries](/influxdb/cloud-serverless/reference/client-libraries/v3/)
to help make your code for writing and querying cross-compatible with InfluxDB Cloud Serverless and Cloud Dedicated.
You'll only need to change your InfluxDB connection credentials
(host, database name, and token).
```

### Avoid features that are not included with InfluxDB Cloud Dedicated

The easiest way to avoid using features in InfluxDB Cloud Serverless that don’t
exist in Cloud Dedicated is to avoid using the Cloud Serverless UI, except when
managing tokens and buckets.

> In order to maintain compatibility with Cloud Dedicated, avoid using the following
> InfluxDB Cloud Serverless features: 

Reviewer: ```suggestion
In order to maintain compatibility with Cloud Dedicated, specifically avoid using the following
InfluxDB Cloud Serverless features: 
```

- The v2 query API and the Flux language
- Administrative APIs
- Tasks and alerts from the Cloud Serverless UI (instead use one of the options
mentioned in _[Tasks and alerts differences](#tasks-and-alerts-differences)_).

> - InfluxDB dashboards and visualization tools (use 3rd party Visualization Tools)

Reviewer: ```suggestion
- InfluxDB dashboards and visualization tools (use 3rd-party visualization tools)
```

### Use SQL or InfluxQL as your Query Language

> Both SQL and InfluxQL are optimized for InfluxDB v3 and both are excellent

Reviewer: ```suggestion
SQL and InfluxQL are optimized for InfluxDB v3 and both are excellent
```

options in Cloud Dedicated and Cloud Serverless.
Avoid Flux since it can’t be used with InfluxDB Cloud Dedicated.

### Stay within the schema limits of InfluxDB Cloud Serverless

If you stay within InfluxDB Cloud Serverless limits for tables (measurements)
and columns (time, fields, and tags) within a table, then you won’t have any
problems with limits in InfluxDB Cloud Dedicated.
Cloud Dedicated also provides more flexibility by letting you configure limits. 

| Description                  | Limit |
| :--------------------------- | ----: |
| Tables (measurements)        |   500 |
| Columns (time, fields, tags) |   200 |

### Keep test and production data separate

If you are using InfluxDB Cloud Serverless for prototyping or evaluation, use
test data and don’t store production data on the platform.
This has the following benefits:

> - If you make a mistake with your schema and you want to start over, it is easier
>   to delete your test data and start from scratch. If you are using production
>   data that you rely on and only have one copy, schema mistakes are harder to
>   recover from.
> - After you have completed your evaluation or prototype, you won’t have any data
>   that needs to be migrated to InfluxDB Cloud Dedicated.

Reviewer: ```suggestion
- Using test data is less risky. If you make a mistake with your schema, you can delete your data and start over. If you use production
  data that you rely on and only have one copy, schema mistakes are harder to
  recover from.
- You won't need to migrate data. After you have completed your evaluation, you can delete your test data and start using Cloud Dedicated.
```

> There is nothing preventing you from using live or real production data in this circumstance.
> You can move data from InfluxDB Cloud Serverless to Cloud Dedicated after you
> have completed your prototype and evaluation. However, currently this require
> that you write code to query and extract the data, convert it to line protocol
> format, and then write it back to Cloud Dedicated.

Reviewer: ```suggestion
_You can use live or production data for protoype in InfluxDB Cloud Serverless, and then migrate it to Cloud Dedicated after you complete your evaluation. However, currently this requires
that you write code to query and extract the data, convert it to line protocol
format, and then write it to InfluxDB Cloud Dedicated._
```
